File: BottomActivity/app/src/main/java/com/example/bottomactivity/MainActivity.java
package com.example.bottomactivity;

import android.content.Intent;
import android.os.Bundle;
import android.support.constraint.ConstraintLayout;
import android.support.design.widget.BottomNavigationView;
import android.support.v7.app.AppCompatActivity;
import android.support.annotation.NonNull;
import android.view.MenuItem;
import android.view.View;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.LinearLayout;
import android.widget.RadioGroup;
import android.widget.TextView;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {
    private LinearLayout homeLayout;
    private LinearLayout bmiLayout;
    private LinearLayout dietLayout;
    private ConstraintLayout imageLayout;
    private int gender;
    private ArrayList<String> statistics = new ArrayList<>();
    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_home:
                    homeLayout.setVisibility(View.VISIBLE);
                    bmiLayout.setVisibility(View.INVISIBLE);
                    dietLayout.setVisibility(View.INVISIBLE);
                    imageLayout.setVisibility(View.INVISIBLE);
                    final TextView welcomeTextView = findViewById(R.id.welcomeTextView);
                    welcomeTextView.setText(R.string.welcome);
                    if (statistics.size() > 0) {
                        findViewById(R.id.chartButton).setOnClickListener(new View.OnClickListener() {
                            @Override
                            public void onClick(View view) {
                                showChart();
                            }
                        });
                    }
                    return true;
                case R.id.navigation_dashboard:
                    homeLayout.setVisibility(View.INVISIBLE);
                    bmiLayout.setVisibility(View.VISIBLE);
                    dietLayout.setVisibility(View.INVISIBLE);
                    imageLayout.setVisibility(View.INVISIBLE);
                    final EditText heightEditText = (EditText) findViewById(R.id.heightEditText);
                    final EditText weightEditText = (EditText) findViewById(R.id.weightEditText);
                    final EditText ageEditText = (EditText) findViewById(R.id.ageEditText);
                    final RadioGroup genderRadioGroup = (RadioGroup) findViewById(R.id.genderRadioGroup);
                    genderRadioGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
                        @Override
                        public void onCheckedChanged(RadioGroup group, int checkedId) {
                            View radioButton = genderRadioGroup.findViewById(checkedId);
                            gender = genderRadioGroup.indexOfChild(radioButton);
                        }
                    });
                    findViewById(R.id.callculateButton).setOnClickListener(new View.OnClickListener() {
                        public void onClick(View v) {
                            if (heightEditText.getText().toString().isEmpty()) {
                                heightEditText.setError("Enter height");
                                heightEditText.requestFocus();
                                return;
                            }
                            if (weightEditText.getText().toString().isEmpty()) {
                                weightEditText.setError("Enter weight");
                                weightEditText.requestFocus();
                                return;
                            }
                            if (ageEditText.getText().toString().isEmpty()) {
                                ageEditText.setError("Enter age");
                                ageEditText.requestFocus();
                                return;
                            }
                            float bmiResult = calculateBMI(Float.parseFloat(weightEditText.getText().toString()), Float.parseFloat(heightEditText.getText().toString()) / 100);
                            bmiResultDetails(bmiResult);
                            float ppmResult = calculatePPM(Float.parseFloat(weightEditText.getText().toString()), Float.parseFloat(heightEditText.getText().toString()), Integer.parseInt(ageEditText.getText().toString()));
                            ppmResultDetails(ppmResult);
                        }
                    });
                    return true;
                case R.id.navigation_notifications:
                    final TextView resultTextView = (TextView) findViewById(R.id.ppmResultTextView);
                    resultTextView.setVisibility(View.VISIBLE);
                    homeLayout.setVisibility(View.INVISIBLE);
                    bmiLayout.setVisibility(View.INVISIBLE);
                    dietLayout.setVisibility(View.VISIBLE);
                    imageLayout.setVisibility(View.VISIBLE);
                    return true;
            }
            return false;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        BottomNavigationView navView = findViewById(R.id.nav_view);
        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);
        if (statistics.size() > 0) {
            findViewById(R.id.chartButton).setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    showChart();
                }
            });
        }
        findViewById(R.id.quizButton).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startQuiz();
            }
        });
        homeLayout = findViewById(R.id.homeLayout);
        bmiLayout = findViewById(R.id.bmiLayout);
        dietLayout = findViewById(R.id.dietLayout);
        imageLayout = findViewById(R.id.imageLayout);
        homeLayout.setVisibility(View.VISIBLE);
        bmiLayout.setVisibility(View.INVISIBLE);
        dietLayout.setVisibility(View.INVISIBLE);
        imageLayout.setVisibility(View.INVISIBLE);
    }

    private void bmiResultDetails(float bmiResult) {
        final TextView resultDetailsTextView = (TextView) findViewById(R.id.bmiResultDetailsTextView);
        final TextView bmiResultTextView = (TextView) findViewById(R.id.bmiResultTextView);
        bmiResultTextView.setText(String.format("%.2f", bmiResult));
        if (bmiResult < 18.5)
            resultDetailsTextView.setText(R.string.bmi_underweight);
        else if (bmiResult >= 18.5 && bmiResult <= 25)
            resultDetailsTextView.setText(R.string.bmi_normalweight);
        else if (bmiResult > 25 && bmiResult <= 30)
            resultDetailsTextView.setText(R.string.bmi_overweight);
        else
            resultDetailsTextView.setText(R.string.bmi_obese);
    }

    private void ppmResultDetails(float ppmResult) {
        final ImageView very_low = findViewById(R.id.veryLowImageView);
        final ImageView low = findViewById(R.id.lowImageView);
        final ImageView normal = findViewById(R.id.normalImageView);
        final ImageView high = findViewById(R.id.highImageView);
        final TextView ppmResultTextView = (TextView) findViewById(R.id.ppmResultTextView);
        ppmResultTextView.setText(String.format("%.2f", ppmResult));
        if (ppmResult < 1600) {
            very_low.findViewById(R.id.veryLowImageView).setVisibility(View.VISIBLE);
            low.findViewById(R.id.lowImageView).setVisibility(View.INVISIBLE);
            normal.findViewById(R.id.normalImageView).setVisibility(View.INVISIBLE);
            high.findViewById(R.id.highImageView).setVisibility(View.INVISIBLE);
        } else if (ppmResult >= 1600 && ppmResult < 1800) {
            very_low.findViewById(R.id.veryLowImageView).setVisibility(View.INVISIBLE);
            low.findViewById(R.id.lowImageView).setVisibility(View.VISIBLE);
            normal.findViewById(R.id.normalImageView).setVisibility(View.INVISIBLE);
            high.findViewById(R.id.highImageView).setVisibility(View.INVISIBLE);
        } else if (ppmResult >= 1800 && ppmResult < 2000) {
            very_low.findViewById(R.id.veryLowImageView).setVisibility(View.INVISIBLE);
            low.findViewById(R.id.lowImageView).setVisibility(View.INVISIBLE);
            normal.findViewById(R.id.normalImageView).setVisibility(View.VISIBLE);
            high.findViewById(R.id.highImageView).setVisibility(View.INVISIBLE);
        } else if (ppmResult >= 2000) {
            very_low.findViewById(R.id.veryLowImageView).setVisibility(View.INVISIBLE);
            low.findViewById(R.id.lowImageView).setVisibility(View.INVISIBLE);
            normal.findViewById(R.id.normalImageView).setVisibility(View.INVISIBLE);
            high.findViewById(R.id.highImageView).setVisibility(View.VISIBLE);
        }
    }

    private float calculateBMI(float weight, float height) {
        statistics.add(String.valueOf(weight));
        return (float) (weight / (Math.pow(height, 2)));
    }

    private float calculatePPM(float weight, float height, int age) {
        if (gender == 0) {
            return (float) ((10 * weight) + (height * 6.25) - (5 * age) - 161);
        } else {
            return (float) ((10 * weight) + (height * 6.25) - (5 * age) + 5);
        }
    }

    private void startQuiz() {
        Intent intent = new Intent(this, QuizActivity.class);
        startActivity(intent);
    }

    private void showChart() {
        Intent intent = new Intent(this, ChartActivity.class);
        intent.putStringArrayListExtra("statistics", statistics);
        startActivity(intent);
    }
}

File: BmiCalculatorKotlin/app/src/main/java/com/app/MainActivity.kt
package com.app

import android.content.Intent
import android.os.Bundle
import android.view.MenuItem
import android.view.View
import android.widget.*
import androidx.annotation.NonNull
import com.google.android.material.bottomnavigation.BottomNavigationView
import androidx.appcompat.app.AppCompatActivity
import androidx.constraintlayout.widget.ConstraintLayout

import java.util.ArrayList
import kotlin.math.pow

class MainActivity : AppCompatActivity() {
    private var homeLayout: LinearLayout? = null
    private var bmiLayout: LinearLayout? = null
    private var dietLayout: LinearLayout? = null
    private var imageLayout: ConstraintLayout? = null
    private var gender: Int = 0
    private val statistics = ArrayList<String>()
    private val mOnNavigationItemSelectedListener =
        object : BottomNavigationView.OnNavigationItemSelectedListener {

            override fun onNavigationItemSelected(@NonNull item: MenuItem): Boolean {
                when (item.itemId) {
                    R.id.navigation_home -> {
                        homeLayout?.visibility = View.VISIBLE
                        bmiLayout?.visibility = View.INVISIBLE
                        dietLayout?.visibility = View.INVISIBLE
                        imageLayout?.visibility = View.INVISIBLE
                        val welcomeTextView: TextView = findViewById(R.id.welcomeTextView)
                        welcomeTextView.setText(R.string.welcome)
                        if (statistics.size > 0) {
                            val chartButton: Button = findViewById(R.id.chartButton)
                            chartButton.setOnClickListener { showChart() }
                        }
                        return true
                    }
                    R.id.navigation_dashboard -> {
                        homeLayout?.visibility = View.INVISIBLE
                        bmiLayout?.visibility = View.VISIBLE
                        dietLayout?.visibility = View.INVISIBLE
                        imageLayout?.visibility = View.INVISIBLE
                        val heightEditText = findViewById<EditText>(R.id.heightEditText)
                        val weightEditText = findViewById<EditText>(R.id.weightEditText)
                        val ageEditText = findViewById<EditText>(R.id.ageEditText)
                        val genderRadioGroup = findViewById<RadioGroup>(R.id.genderRadioGroup)
                        genderRadioGroup.setOnCheckedChangeListener { _, checkedId ->
                            val radioButton = genderRadioGroup.findViewById<View>(checkedId)
                            gender = genderRadioGroup.indexOfChild(radioButton)
                        }
                        val callculateButton: Button = findViewById(R.id.callculateButton)
                        callculateButton.setOnClickListener(View.OnClickListener {
                            if (heightEditText.text.toString().isEmpty()) {
                                heightEditText.error = "Enter height"
                                heightEditText.requestFocus()
                                return@OnClickListener
                            }
                            if (weightEditText.text.toString().isEmpty()) {
                                weightEditText.error = "Enter weight"
                                weightEditText.requestFocus()
                                return@OnClickListener
                            }
                            if (ageEditText.text.toString().isEmpty()) {
                                ageEditText.error = "Enter age"
                                ageEditText.requestFocus()
                                return@OnClickListener
                            }
                            val bmiResult = calculateBMI(
                                java.lang.Float.parseFloat(weightEditText.text.toString()),
                                java.lang.Float.parseFloat(heightEditText.text.toString()) / 100
                            )
                            bmiResultDetails(bmiResult)
                            val ppmResult = calculatePPM(
                                java.lang.Float.parseFloat(weightEditText.text.toString()),
                                java.lang.Float.parseFloat(heightEditText.text.toString()),
                                Integer.parseInt(ageEditText.text.toString())
                            )
                            ppmResultDetails(ppmResult)
                        })
                        return true
                    }
                    R.id.navigation_notifications -> {
                        val resultTextView = findViewById<TextView>(R.id.ppmResultTextView)
                        resultTextView.visibility = View.VISIBLE
                        homeLayout?.visibility = View.INVISIBLE
                        bmiLayout?.visibility = View.INVISIBLE
                        dietLayout?.visibility = View.VISIBLE
                        imageLayout?.visibility = View.VISIBLE
                        return true
                    }
                }
                return false
            }
        }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val navView: BottomNavigationView = findViewById(R.id.nav_view)
        navView.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)
        if (statistics.size > 0) {
            val chartButton: Button = findViewById(R.id.chartButton)
            chartButton.setOnClickListener { showChart() }
        }
        val quizButton: Button = findViewById(R.id.quizButton)
        quizButton.setOnClickListener { startQuiz() }
        homeLayout = findViewById(R.id.homeLayout)
        bmiLayout = findViewById(R.id.bmiLayout)
        dietLayout = findViewById(R.id.dietLayout)
        imageLayout = findViewById(R.id.imageLayout)
        homeLayout?.visibility = View.VISIBLE
        bmiLayout?.visibility = View.INVISIBLE
        dietLayout?.visibility = View.INVISIBLE
        imageLayout?.visibility = View.INVISIBLE
    }

    private fun bmiResultDetails(bmiResult: Float) {
        val resultDetailsTextView = findViewById<TextView>(R.id.bmiResultDetailsTextView)
        val bmiResultTextView = findViewById<TextView>(R.id.bmiResultTextView)
        bmiResultTextView.text = String.format("%.2f", bmiResult)
        if (bmiResult < 18.5)
            resultDetailsTextView.setText(R.string.bmi_underweight)
        else if (bmiResult in 18.5..25.0)
            resultDetailsTextView.setText(R.string.bmi_normalweight)
        else if (bmiResult > 25 && bmiResult <= 30)
            resultDetailsTextView.setText(R.string.bmi_overweight)
        else
            resultDetailsTextView.setText(R.string.bmi_obese)
    }

    private fun ppmResultDetails(ppmResult: Float) {
        val veryLow: ImageView = findViewById(R.id.veryLowImageView)
        val low: ImageView = findViewById(R.id.lowImageView)
        val normal: ImageView = findViewById(R.id.normalImageView)
        val high: ImageView = findViewById(R.id.highImageView)
        val ppmResultTextView = findViewById<TextView>(R.id.ppmResultTextView)
        ppmResultTextView.text = String.format("%.2f", ppmResult)
        if (ppmResult < 1600) {
            veryLow.visibility = View.VISIBLE
            low.visibility = View.INVISIBLE
            normal.visibility = View.INVISIBLE
            high.visibility = View.INVISIBLE
        } else if (ppmResult >= 1600 && ppmResult < 1800) {
            veryLow.visibility = View.INVISIBLE
            low.visibility = View.VISIBLE
            normal.visibility = View.INVISIBLE
            high.visibility = View.INVISIBLE
        } else if (ppmResult >= 1800 && ppmResult < 2000) {
            veryLow.visibility = View.INVISIBLE
            low.visibility = View.INVISIBLE
            normal.visibility = View.VISIBLE
            high.visibility = View.INVISIBLE
        } else if (ppmResult >= 2000) {
            veryLow.visibility = View.INVISIBLE
            low.visibility = View.INVISIBLE
            normal.visibility = View.INVISIBLE
            high.visibility = View.VISIBLE
        }
    }

    private fun calculateBMI(weight: Float, height: Float): Float {
        statistics.add(weight.toString())
        return (weight / height.toDouble().pow(2.0)).toFloat()
    }

    private fun calculatePPM(weight: Float, height: Float, age: Int): Float {
        return if (gender == 0) {
            (10 * weight + height * 6.25 - (5 * age).toDouble() - 161.0).toFloat()
        } else {
            (10 * weight + height * 6.25 - 5 * age + 5).toFloat()
        }
    }

    private fun startQuiz() {
        val intent = Intent(this, QuizActivity::class.java)
        startActivity(intent)
    }

    private fun showChart() {
        val intent = Intent(this, ChartActivity::class.java)
        intent.putStringArrayListExtra("statistics", statistics)
        startActivity(intent)
    }
}
